service/README.md

```diff
@@ -42,6 +42,35 @@ automatically load it, and include it in its normal annotation processing
 environment. Other users of java.util.ServiceLoader may use the infrastructure
 to different ends, but this metadata will provide auto-loading appropriately.
 
+When implementing standalone Gradle Plugins you can use `@AutoService` along with `@PluginName` in order
+to register implementations of well-known types using META-INF metadata.
+
+## Example
+
+Say you have:
+
+```java
+package foo.bar;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+
+@AutoService(Plugin.class)
+@PluginName("com.example")
+final class MyPlugin implements Plugin<Project> {
+  // …
+}
+```
+
+AutoService will generate the file
+`META-INF/gradle-plugins/com.example.properties` in the output classes
```

Comment:
Since AutoService sort of lands things in `META-INF/services`, would it be better to have an `@AutoGradlePlugin` annotation / corresponding processor ?

Reply:
As a matter of fact it actually makes sense.

I am not very good with names and `PluginName` does not really express the name but rather the id. My original thought was to name the annotation `PluginId` but because Gradle already has classes named as `Plugin` and `PluginId` naming the annotation as such would make it confusing for having to import the fully qualified name every time.